app/controllers/admins/calculators_controller.rb

```diff
@@ -56,11 +56,17 @@ def calculator
       @calculator = Calculator.friendly.find(params[:slug])
     end
 
-    def collect_form_fields
-      @all_fields = @calculator.fields
-      @form_fields = @calculator.fields.form.order(:created_at)
-      @parameter_fields = @calculator.fields.parameter.order(:created_at)
-      @result_fields = @calculator.fields.result.order(:created_at)
+    def collect_fields_for_form
+      @form_fields = collect_fields_for_kind('form')
+      @parameter_fields = collect_fields_for_kind('parameter')
+      @result_fields = collect_fields_for_kind('result')
+    end
+
+    def collect_fields_for_kind(kind)
+      @calculator
+        .fields
+        .select { |field| field.kind == kind }
```

Comment:
`select` -> `where(kind: kind)`

Reply:
If I change it, nested fields won't show errors